org.adempiere.base/src/org/compiere/model/PO.java

```diff
@@ -6223,5 +6240,209 @@ public boolean columnExists(String columnName, boolean throwException) {
+						}
+						else
+						{
+							value = rs.getString(4);
+						}
+					}
+					else
+					{
+						value = rs.getString(4);
+					}
+
+					if (value != null)
+						attributeMap.put(attName, value);
+				}
+			}
+			catch (Exception e)
+			{
+				CLogger.get().log(Level.SEVERE, "Failed: Get Attribute = " + attributeName, e);
+				return null;
+			}
+			finally
+			{
+				DB.close(rs, pstmt);
+				rs = null;
+				pstmt = null;
+			}
+		}
+
+		if (attributeMap.containsKey(attributeName))
+			return attributeMap.get(attributeName);
+
+		return getAttributeDefaultValue(attributeName, get_Table_ID());
+	} // getAttribute
+
+	/**
+	 * Return attribute default value for table.
+	 *  
+	 * @param  attributeName
+	 * @param tableID 
+	 * @param  table_ID
+	 * @return
+	 */
+	private static Object getAttributeDefaultValue(String attributeName, int tableID)
+	{
+		String key = tableID + "_" + attributeName;
+		if (!s_tableAttributeDefault.containsKey(key))
+		{
+			PreparedStatement pstmt = null;
+			ResultSet rs = null;
+			Object value = null;
+			try
+			{
+				pstmt = DB.prepareStatement(TABLE_ATTRIBUTE_DEFAULTVALUE_SQL, null);
+				pstmt.setString(1, attributeName);
+				pstmt.setInt(2, tableID);
+				rs = pstmt.executeQuery();
+				if(rs.next())
+				{
+					String attType = rs.getString(2);
+					int reference_ID = rs.getInt(3);
+					String DefaultValue = rs.getString(4);
+
+					if (Util.isEmpty(DefaultValue))
+						return null;
+
+					if (MAttribute.ATTRIBUTEVALUETYPE_Number.equalsIgnoreCase(attType)
+						|| MAttribute.ATTRIBUTEVALUETYPE_List.equalsIgnoreCase(attType))
+					{
+						value = Integer.valueOf(DefaultValue);
+					}
+					else if (MAttribute.ATTRIBUTEVALUETYPE_Date.equalsIgnoreCase(attType))
+					{
+						value = Timestamp.valueOf(DefaultValue);
+					}
+					else if (MAttribute.ATTRIBUTEVALUETYPE_StringMax40.equalsIgnoreCase(attType))
+					{
+						value = DefaultValue;
+					}
+					else if (MAttribute.ATTRIBUTEVALUETYPE_Reference.equalsIgnoreCase(attType))
+					{
+						if (DisplayType.isText(reference_ID) || reference_ID == DisplayType.YesNo)
+						{
+							value = DefaultValue;
+						}
+						else if (DisplayType.isDate(reference_ID))
+						{
+							value = Timestamp.valueOf(DefaultValue);
+						}
+						else if (DisplayType.isNumeric(reference_ID) || DisplayType.isID(reference_ID))
+						{
+							value = Integer.valueOf(DefaultValue);
+						}
+						else
+						{
+							value = DefaultValue;
+						}
+					}
+					else
+					{
+						value = DefaultValue;
+					}
+					s_tableAttributeDefault.put(key, value);
+				}
+			}
+			catch (Exception e)
+			{
+				CLogger.get().log(Level.SEVERE, "Failed: Get Attribute = " + attributeName, e);
+				return null;
+			}
+			finally
+			{
+				DB.close(rs, pstmt);
+				rs = null;
+				pstmt = null;
+			}
+		}
+
+		if (s_tableAttributeDefault.containsKey(key))
+			return s_tableAttributeDefault.get(key);
+		return null;
+	} // getAttributeDefaultValue
```

Comment:
Going by the current practise in iDempiere code, I think it is more appropriate to add this new attributes related method to MTableAttribute instead.